lib/destroy.js

```diff
@@ -30,13 +30,9 @@ async function destroy() {
     }
 }
 
-function makeWorkspaceOrNot(createworkspace, workspace) {
-    if (createworkspace != "false") {
-        return [`workspace`, `new`, `${workspace}`]
-    }
-    return [`workspace`, `select`, `${workspace}`]
+function makeWorkspaceIfNotExits(execfile, workspace) {
+    return [`${execfile} workspace select ${workspace} || ${execfile} workspace new ${workspace}`]
```

Comment:
I like the way we are heading here. I agreed that the creation of the workspace shouldn't fail if it already exists, and the code should be smart enough to handle it, this is the correct behaviour. I do think, however, we might be able to improve this by: (1) Not even attempt to create the workspace if it already exists and print a message sinalizing that to the user and (2) Create in case it doesn't. 

Perhaps, something like:

```javascript
if (createworkspace) {
    // Let's only care about checking the existence of the workspace, after we know the
    // the user wants the workspace to be created. After all, listing the workspace is little
    // expensive.
    workspace_exists = run("terraform workspace list | grep -q prod")
    if (!workspace_exists) {
       core.info("Workspace ${workspace} does't not exist, creating it")
        ... create workspace command ...
    }

    ... select workspace ...
}
```

This will not change only the function code, but the selection and/or creation of the workspace are also outsource to this function. 

We could also, return different values to the caller function with the "create && select" or just select. Something like:

```javascript
if (createworkspace) {
    // Let's only care about checking the existence of the workspace, after we know the
    // the user want the workspace to be created. Specially, because this operation is a bit
    // expensive. 
    workspace_exists = run("terraform workspace list | grep -q prod")
    if (!workspace_exists) {
       core.info("Workspace ${workspace} does't not exist, creating it")
        return ["terraform workspace new ${workspace} && terraform workspace select ${workspace}"]
    }

    return ["terraform workspace select ${workspace}"]
}
```

WDT?

In terms of implementation, I prefer the first because the command execution will be outsource to the function and we call in all terraform stages. Currently, we have to call the function and exec in most of the terraform step commands.

Reply:
HI, @raffs, Thanks for the suggestions I try to work on similar lines,  can you please review the latest changes if this will work for us?